docs(google-cloud): add Workload Identity Federation authentication (#1637)
seqeralabs/docs: platform-cloud/docs/compute-envs/google-cloud.md

If your Google Cloud project does not require access restrictions on any of its Cloud Storage buckets, you can grant project Storage Admin (`roles/storage.admin`) permissions to your service account to simplify setup. To grant access only to specific buckets, add the service account as a principal [on each bucket individually](https://docs.seqera.io/platform-cloud/compute-envs/google-cloud-batch#cloud-storage-bucket). For each Google Cloud compute environment created in the Seqera platform, a separate service account is created with the necessary permissions to launch pipelines/studios.

:::caution

> On shared GCP projects, `roles/resourcemanager.projectIamAdmin` allows the service account to grant any role to any principal on the project — a compromised credential can escalate to any project-level role. Similarly, `roles/iam.serviceAccountAdmin` grants create and delete access to any service account in the project. As an advanced hardening option, add an [IAM condition](https://cloud.google.com/iam/docs/conditions-overview) to the `roles/iam.serviceAccountAdmin` binding to restrict it to service accounts whose names start with `towerforge-`.

Reviewer: ```suggestion
On shared GCP projects, `roles/resourcemanager.projectIamAdmin` lets the service account grant any role to any principal on the project. A compromised credential can then escalate to any project-level role. `roles/iam.serviceAccountAdmin` grants create and delete access to any service account in the project.

To harden this, add an [IAM condition](https://cloud.google.com/iam/docs/conditions-overview) to the `roles/iam.serviceAccountAdmin` binding that restricts it to service accounts whose names start with `towerforge-`.
```

:::

#### Userdata script error detection (optional)

Platform can retrieve the serial port output of the Compute Engine instance to detect errors in the userdata script that bootstraps the VM during instance startup. This capability is included in the `roles/compute.instanceAdmin.v1` role listed above. If you use a custom role instead, include the `compute.instances.getSerialPortOutput` permission. Without this permission, userdata script failures are not detected, and no warning is shown.

### Authentication methods

Seqera supports two methods for authenticating with Google Cloud:

#### Service account keys

To authenticate using a service account key, create a [service account JSON key file](https://cloud.google.com/iam/docs/keys-list-get#get-key):

1. In the Google Cloud navigation menu, select **IAM & Admin > Service Accounts**.
2. Select the email address of the service account.
3. Select **Keys > Add key > Create new key**.
4. Select **JSON** as the key type.
5. Select **Create**.

> A JSON file is downloaded to your computer. This file contains the credential needed to configure the compute environment in Seqera.

Reviewer: ```suggestion
Google Cloud downloads a JSON file to your computer. It contains the credential you need to configure the compute environment in Seqera.
```

#### Workload Identity Federation

Workload Identity Federation (WIF) is the recommended authentication method for production and regulated environments because it eliminates the need for long-lived service account keys. WIF uses short-lived OIDC tokens for authentication, which are generated by Seqera Platform.

> Platform's OIDC issuer is the issuer value advertised at `https://cloud.seqera.io/api/.well-known/openid-configuration`.

Reviewer: ```suggestion
Workload Identity Federation (WIF) is the recommended authentication method for production and regulated environments because it avoids long-lived service account keys. Instead, Seqera Platform generates short-lived OIDC tokens for authentication.

Platform advertises its OIDC issuer at https://cloud.seqera.io/api/.well-known/openid-configuration.
```

##### Enable APIs

Before setting up WIF, enable the following APIs for your Google Cloud project:

- [Cloud Resource Manager API](https://console.cloud.google.com/marketplace/product/google/cloudresourcemanager.googleapis.com) (`cloudresourcemanager.googleapis.com`)
- [IAM API](https://console.cloud.google.com/marketplace/product/google/iam.googleapis.com) (`iam.googleapis.com`)
- [IAM Service Account Credentials API](https://console.cloud.google.com/marketplace/product/google/iamcredentials.googleapis.com) (`iamcredentials.googleapis.com`)
- [Cloud Logging API](https://console.cloud.google.com/marketplace/product/google/logging.googleapis.com) (`logging.googleapis.com`)
- [Security Token Service API](https://console.cloud.google.com/marketplace/product/google/sts.googleapis.com) (`sts.googleapis.com`)

:::note
The Compute Engine, Cloud Storage, and Secret Manager APIs (`compute.googleapis.com`, `storage.googleapis.com`, `secretmanager.googleapis.com`) are also required. Enable them if not already active in your project.
:::

##### Set up WIF in the GCP Console

> Setting up WIF requires the following steps in the GCP Console:

Reviewer: ```suggestion
In the GCP Console:
```

1. Create a [Workload Identity Pool and Provider](https://cloud.google.com/iam/docs/workload-identity-federation-with-other-providers) in your Google Cloud project.
2. Set Seqera as an OIDC provider within the pool. Set the **Issuer URL** to `https://cloud.seqera.io/api`.
3. Set the **Allowed audiences**. If left empty, GCP derives a default audience from the provider resource path in the format `//iam.googleapis.com/projects/{PROJECT}/locations/global/workloadIdentityPools/{POOL}/providers/{PROVIDER}`. If you specify a custom value, it must match exactly what you enter in the **Token audience** field when creating the WIF credential in Seqera.

> 4. Define an attribute mapping and condition. At a minimum, set `google.subject=assertion.sub`. This maps the subject claim from Seqera's JWT to GCP's identity space. For more information, see [Attribute mappings and conditions](https://cloud.google.com/iam/docs/workload-identity-federation-with-other-providers#mappings-and-conditions). You may see a pop-up asking to configure your application and provide an OIDC ID token path — dismiss it.

Reviewer: ```suggestion
4. Define an attribute mapping and condition. At a minimum, set google.subject=assertion.sub. This maps the subject claim from Seqera's JWT to GCP's identity space. For more information, see [Attribute mappings and conditions](https://cloud.google.com/iam/docs/workload-identity-federation-with-other-providers#mappings-and-conditions). If a pop-up asks you to configure your application and provide an OIDC ID token path, dismiss it.
```
